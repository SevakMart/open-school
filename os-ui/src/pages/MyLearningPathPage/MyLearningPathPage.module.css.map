{"version":3,"sources":["MyLearningPathPage.module.scss"],"names":[],"mappings":"AAIA;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;EAAA","file":"MyLearningPathPage.module.css"}