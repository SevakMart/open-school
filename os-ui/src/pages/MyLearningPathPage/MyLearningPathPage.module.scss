@import "../../styles/Mixins/ButtonMixin.scss";
@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";

.mainHeader {
    @include fontStyle('Montserrat',normal,800,4rem,3.688rem);
    color:#000000;
    margin-top:3%;
    padding-left:6%
}

.courseNavigationBar{
    @include flexStyle(row,center,space-between);
    padding-left:6%;
    width:15%;
    .activeNav{
        cursor:pointer;
        @include fontStyle('Montserrat',normal,600,1.25rem,1.5rem);
        color:#222427;
        border-bottom: 2px solid #222427 ;
    }
    .nonActiveNav {
        @include fontStyle('Montserrat',normal,500,1rem,1.5rem);
        color:#5E617B;
        cursor:pointer;
        &:hover {
            color:#222427;
        }
    }
}
.courseContainer {
    @include flexStyle(column,center,space-between);
}
.suggestedCoursesTitle{
    @include fontStyle('Montserrat',normal,500,1rem,1.5rem);
    color:#5E617B;
    margin-top:5%;
    padding-left:6%;
}
.suggestedCoursesContainer{
    @include flexStyle(row,center,space-between);
    padding-left:6%;
    padding-right:6%;
}