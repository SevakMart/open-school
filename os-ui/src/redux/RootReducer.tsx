import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
