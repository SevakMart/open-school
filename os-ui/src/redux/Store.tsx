import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
