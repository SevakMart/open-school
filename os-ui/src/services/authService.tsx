import fetchService from './fetchData';
import { RegistrationFormType } from '../types/RegistartionFormType';
import { ResetPasswordType } from '../types/ResetPasswordType';

class AuthService {
  readonly basePath: string;

  constructor() {
    this.basePath = 'auth';
  }

  async register(content:RegistrationFormType) {
    const response = await fetchService.post(`${this.basePath}/register`, content, {});
    const { status } = response;
    const data = await response.json();
    return { ...data, status };
  }

  async signIn(content:Omit<RegistrationFormType, 'firstName' | 'lastName'>) {
    const response = await fetchService.post(`${this.basePath}/login`, content, {});
    const { status, headers } = response;
    const data = await response.json();
    const FullTokenResponse = headers.get('Authorization');
    const tokenStartingIndex = FullTokenResponse ? FullTokenResponse.indexOf(' ') : undefined;
    const token = FullTokenResponse && tokenStartingIndex
      ? FullTokenResponse.slice(tokenStartingIndex + 1) : null;
    return { ...data, status, token };
  }

  async redirectUserToAccount(content:object) {
    const response = await fetchService.post(`${this.basePath}/account`, content, {});
    const { status, headers } = response;
    let data = {};
    if (response.status === 200) {
      data = await response.json();
    }
    const FullTokenResponse = headers.get('Authorization');
    const tokenStartingIndex = FullTokenResponse ? FullTokenResponse.indexOf(' ') : undefined;
    const token = FullTokenResponse && tokenStartingIndex
      ? FullTokenResponse.slice(tokenStartingIndex + 1) : null;
    return { ...data, status, token };
  }

  async sendForgotPasswordRequest(content:object) {
    const response = await fetchService.post(`${this.basePath}/password/forgot`, content, {});
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async sendResetPasswordRequest(content:ResetPasswordType) {
    const response = await fetchService.post(`${this.basePath}/password/reset`, content, {});
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }
}

export default new AuthService();
