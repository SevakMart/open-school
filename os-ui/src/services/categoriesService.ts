import { fetchDataGet } from './fetchData';

class CategoriesService {
  readonly baseUrl: string;

  constructor() {
    this.baseUrl = '/api/v1/categories';
  }

  getCategories(params: object = {}) {
    return fetchDataGet(this.baseUrl, params);
  }
}

export default new CategoriesService();
