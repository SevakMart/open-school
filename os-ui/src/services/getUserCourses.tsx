import { fetchData } from './fetchData';

export const getUserCourses = async (url:string) => {
  const data = await fetchData(url);
  return data;
};
