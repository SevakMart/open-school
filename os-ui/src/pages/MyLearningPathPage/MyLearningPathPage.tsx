import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { RootState } from '../../redux/Store';
import NavbarOnSignIn from '../../component/NavbarOnSignIn/NavbarOnSignIn';
import NoCourses from './Subcomponents/NoCourses/NoCourses';
import InProgressCourse from './Subcomponents/InProgressCourse/InProgressCourse';
import CompletedCourse from './Subcomponents/CompletedCourse/CompledtedCourse';
import LearningPath from '../../component/LearningPath/LearningPath';
import {
  MY_LEARNING_PATHS, ALL, IN_PROGRESS, COMPLETED,
  SUGGESTED_LEARNING_PATHS, EXPLORE_COURSES,
} from '../../constants/Strings';
import userService from '../../services/userService';
import { UserCourseType } from '../../types/UserCourseType';
import { SuggestedCourseType } from '../../types/SuggestedCourseType';
import styles from './MyLearningPathPage.module.scss';

enum LearningPathNav {
  All='All',
  InProgress='InProgress',
  Completed='Completed'
}

const MyLearningPathPage = ({ userInfo }:any) => {
  const navigate = useNavigate();
  // const userInfo = useSelector<RootState>((state) => state.userInfo);
  const { t } = useTranslation();
  const [userCourses, setUserCourses] = useState<UserCourseType[]>([]);
  const [suggestedCourses, setSuggestedCourses] = useState<SuggestedCourseType[]>([]);
  const [activeNavType, setActiveNavType] = useState(LearningPathNav.All);
  const {
    mainHeader, courseNavigationBar, activeNav, nonActiveNav, courseContainer,
    suggestedCoursesTitle, suggestedCoursesContainer, exploreCourseButton,
  } = styles;

  const handleNavigation = (e:React.SyntheticEvent) => {
    setActiveNavType((e.target as HTMLParagraphElement).dataset.testid as any);
  };

  const exploreLearningPaths = () => {
    navigate('/exploreLearningPaths');
  };

  useEffect(() => {
    let cancel = false;
    if (!suggestedCourses.length) {
      userService.getSuggestedCourses((userInfo as any).id, (userInfo as any).token)
        .then((data) => {
          if (cancel) return;
          if (!data.errorMessage) {
            setSuggestedCourses(data);
          }
        });
    }
    switch (activeNavType) {
      case LearningPathNav.All:
        userService.getUserCourses((userInfo as any).id, (userInfo as any).token)
          .then((data) => {
            if (cancel) return;
            if (!data.errorMessage) {
              setUserCourses(data);
            }
          });
        break;
      case LearningPathNav.InProgress:
        userService.getUserCourses(
          (userInfo as any).id,
          (userInfo as any).token,
          { courseStatusId: 1 },
        )
          .then((data) => {
            if (cancel) return;
            if (!data.errorMessage) {
              setUserCourses(data);
            }
          });
        break;
      case LearningPathNav.Completed:
        userService.getUserCourses(
          (userInfo as any).id,
          (userInfo as any).token,
          { courseStatusId: 2 },
        )
          .then((data) => {
            if (cancel) return;
            if (!data.errorMessage) {
              setUserCourses(data);
            }
          });
        break;
    }
    return () => { cancel = true; };
  }, [activeNavType]);
  return (
    <>
      <NavbarOnSignIn />
      <h1 className={mainHeader}>{MY_LEARNING_PATHS}</h1>
      <nav className={courseNavigationBar}>
        <p className={activeNavType === 'All' ? activeNav : nonActiveNav} data-testid={LearningPathNav.All} onClick={handleNavigation}>{ALL}</p>
        <p className={activeNavType === 'InProgress' ? activeNav : nonActiveNav} data-testid={LearningPathNav.InProgress} onClick={handleNavigation}>{IN_PROGRESS}</p>
        <p className={activeNavType === 'Completed' ? activeNav : nonActiveNav} data-testid={LearningPathNav.Completed} onClick={handleNavigation}>{COMPLETED}</p>
      </nav>
      <div className={courseContainer}>
        {
          userCourses.length === 0 ? <NoCourses />
            : userCourses.map((userCourse, index) => {
              if (userCourse.courseStatus === 'IN_PROGRESS') {
                return (
                  <InProgressCourse
                    key={index}
                    title={userCourse.title}
                    courseStatus={userCourse.courseStatus}
                    percentage={userCourse.percentage}
                    remainingTime={userCourse.remainingTime}
                    dueDate={userCourse.dueDate}
                  />
                );
              }
              return (
                <CompletedCourse
                  key={index}
                  title={userCourse.title}
                  courseStatus={userCourse.courseStatus}
                  grade={userCourse.grade}
                />
              );
            })
        }
        <button type="button" className={exploreCourseButton} onClick={exploreLearningPaths}>{EXPLORE_COURSES}</button>
      </div>
      <p className={suggestedCoursesTitle}>{SUGGESTED_LEARNING_PATHS}</p>
      <div className={suggestedCoursesContainer}>
        {
          suggestedCourses.length ? suggestedCourses.map((suggestedCourse, index) => (
            <LearningPath
              key={index}
              courseInfo={suggestedCourse}
            />
          )) : <h2>{t('No courses.suggested')}</h2>
        }
      </div>
    </>
  );
};
export default MyLearningPathPage;
