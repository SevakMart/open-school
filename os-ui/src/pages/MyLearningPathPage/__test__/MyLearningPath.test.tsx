import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../redux/Store';
import * as fetchUserCourses from '../../../services/getUserCourses';
import MyLearningPathPage from '../MyLearningPathPage';

const userAllCourseData = [
  {
    title: 'React js',
    courseStatus: 'IN_PROGRESS',
    percentage: 47,
    remainingTime: 5400000,
    dueDate: '2022-05-07',
  },
  {
    title: 'Node js',
    courseStatus: 'COMPLETED',
    grade: 90,
  },
];

describe('Create test case for Learning path component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(<Provider store={store}><MyLearningPathPage /></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
  /* test('Check if clicking in_progress will show only in progress courses', async () => {
    expect.hasAssertions();
    const spy = jest.spyOn(fetchUserCourses, 'getUserCourses');
    spy.mockResolvedValue(userAllCourseData);
    render(<Provider store={store}><MyLearningPathPage /></Provider>);
    const inProgressNavItem = screen.queryByTestId('InProgress');
    spy.mockImplementation(()=>userAllCourseData[0]);
    userEvent.click(inProgressNavItem as HTMLParagraphElement);

    const courseTitleElement1 = await screen.findByTestId('React js');
    const courseTitleElement2 = await screen.findByTestId('Node js');

    expect(courseTitleElement1).toBeInTheDocument();
    expect(courseTitleElement2).not.toBeInTheDocument();
  }); */
});
