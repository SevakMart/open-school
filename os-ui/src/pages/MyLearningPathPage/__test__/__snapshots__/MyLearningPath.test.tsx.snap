// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create several unit tests for MyLearningPathPage Component Create a Snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="mainContent"
  >
    <h2>
      string.homePage.navBar.logo
    </h2>
    <div
      class="navMainContent"
    >
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.allLearningPaths
      </p>
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.myLearningPaths
      </p>
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.mentors
      </p>
      <div
        class="bookmarkIconStyle"
      >
        <img
          alt="bookmark"
          src="bookMark.svg"
          style="font-size: 20px; cursor: pointer; fill: none;"
        />
      </div>
      <img
        alt="Notification"
        class="notificationIcon"
        src="Notification.svg"
      />
      <div
        class="userInfoContent"
      >
        <img
          alt="avatar"
          class="avatarLogo"
          src="https://reactjs.org/logo-og.png"
        />
        <img
          alt="Arrow down Icon"
          class="arrowDownIcon"
          src="Arrow-down.svg"
        />
      </div>
      <div />
    </div>
  </nav>
  <h1
    class="mainHeader"
  >
    string.myLearningPaths.mainHeaderTitle
  </h1>
  <nav
    class="courseNavigationBar"
  >
    <p
      class="activeNav"
      data-testid="All"
    >
      string.myLearningPaths.courseStatus.all
    </p>
    <p
      class="nonActiveNav"
      data-testid="InProgress"
    >
      string.myLearningPaths.courseStatus.inProgress
    </p>
    <p
      class="nonActiveNav"
      data-testid="Completed"
    >
      string.myLearningPaths.courseStatus.completed
    </p>
  </nav>
  <div
    class="courseContainer"
  >
    <div
      class="mainContent"
    >
      <div
        class="loader"
      >
        <span />
        <span />
        <span />
      </div>
    </div>
  </div>
  <div
    style="text-align: center;"
  >
    <button
      class="exploreCourseButton"
      type="button"
    >
      button.myLearningPathsPage.exploreCourses
    </button>
  </div>
  <div
    class="mainContainer"
  >
    <p
      class="mainTitle"
    >
      string.myLearningPaths.suggestedLearningPaths
    </p>
    <div
      class="noSuggestedCourses"
    >
      <div
        class="mainContent"
      >
        <div
          class="loader"
        >
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
